packages/optimizer/lib/AmpConstants.js

```diff
@@ -15,10 +15,25 @@
  */
 'mode strict';
 
+const {hasAttribute} = require('./NodeUtils');
+
 module.exports = {
   AMP_TAGS: ['amp', '⚡', '⚡4ads', 'amp4ads', '⚡4email', 'amp4email'],
   AMP_CACHE_HOST: 'https://cdn.ampproject.org',
   AMP_FORMATS: ['AMP', 'AMP4EMAIL', 'AMP4ADS'],
   AMP_RUNTIME_CSS_PATH: '/v0.css',
   appendRuntimeVersion: (prefix, version) => prefix + '/rtv/' + version,
+  isTemplate: (node) => {
+    if (node.tagName === 'template') {
+      return true;
+    }
+    if (
+      node.tagName === 'script' &&
+      hasAttribute(node, 'template') &&
```

Comment:
is there a reason to check for its existence? if it doesn't, the next line would just be against undefined, right?

seems like it could be terse-d up into 

```js
return node.tagName === 'template' || node.tageName === 'script' && node.attribs.template === 'amp-mustache'
```

or even 
```js
return node.tagName === 'template' || node.attribs.template === 'amp-mustache'
```
no worries either way

Reply:
attribs might be undefined